extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -96,22 +123,28 @@ class AmpLightbox extends AMP.BaseElement {
 
     this.isScrollable_ = this.element.hasAttribute('scrollable');
 
-    this.children_ = this.getRealChildren();
+    const children = this.getRealChildren();
 
     this.container_ = this.element.ownerDocument.createElement('div');
     if (!this.isScrollable_) {
       this.applyFillContent(this.container_);
     }
     this.element.appendChild(this.container_);
 
-    this.children_.forEach(child => {
-      if (this.isScrollable_) {
-        this.setAsOwner(child);
-      }
+    children.forEach(child => {
       this.container_.appendChild(child);
     });
 
+    // If scrollable, take ownership of existing children and all future
+    // dynamically created children as well.
     if (this.isScrollable_) {
+      this.takeOwnershipOfDescendants_();
+
+      this.element.addEventListener(AmpEvents.DOM_UPDATE, () => {
+        this.takeOwnershipOfDescendants_();
+        this.updateChildrenInViewport_(this.pos_, this.pos_);
```

Comment:
This means we won't call `updateInViewport` for new children. WAI?

Reply:
We would, on debounced scroll we also calculate visible children and call scheduleLayout and updateInViewport on them.